### PR TITLE
add cue CLI

### PR DESCRIPTION
**Pull request by developer-guy**
Fixes:

Related:

#### For new package PRs only

- [ ] This PR is marked as fixing a pre-existing package request bug
  - [ ] Alternatively, the PR is marked as related to a pre-existing package request bug, such as a dependency
- [ ] REQUIRED - The package is available under an OSI-approved or FSF-approved license
- [ ] REQUIRED - The version of the package is still receiving security updates

#### For security-related PRs

- [ ] The security fix is recorded in `annotations` and `secfixes`

#### For version bump PRs

- [ ] The `epoch` field is reset to 0
- [ ] Patch source: _patch source here_

I used here: _https://github.com/NixOS/nixpkgs/blob/1010c17591db2553d4954cc6a143169604f150e4/pkgs/development/tools/cue/default.nix#L25_

### CODE REVIEW

**Comment by developer-guy** [comment]
lint errors seem not related to this PR 🤔

File: Makefile [hunk 1]
```diff
@@ -79,6 +79,8 @@ PACKAGES := $(addprefix packages/,$(PKGNAMELIST))
 
 $(foreach pkg,$(PKGLIST),$(eval $(call build-package,$(pkg))))
 
+$(eval $(call build-package,cue,0.4.3-r0))
```
Comment (imjasonh) [inline Makefile:82]: We're in the process of changing our build process to not require these updates to the makefile, and instead we'll update `packages.txt`. Can you make that change there so we can build and merge this?

File: cue.yaml [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+  - runs: |
+      CGO_ENABLED=0 go build \
+        -trimpath -ldflags "-s -w -buildid= -X cuelang.org/go/cmd/cue/cmd.version=${{package.version}}" \
+        -o cue-cli ./cmd/cue
+      mkdir -p ${{targets.destdir}}/usr/share/bash-completion/completions 
+      mkdir -p ${{targets.destdir}}/usr/share/zsh/site-functions
+      mkdir -p ${{targets.destdir}}/usr/share/fish/vendor_completions.d
+      ./cue-cli completion bash > ${{targets.destdir}}/usr/share/bash-completion/completions/cue
+      ./cue-cli completion zsh > ${{targets.destdir}}/usr/share/zsh/site-functions/_cue
+      ./cue-cli completion fish > ${{targets.destdir}}/usr/share/fish/vendor_completions.d/cue.fish
```
[the remark this answers is not in the document]
Comment (developer-guy) [reply cue.yaml:31]: What should I do ? 🫡

Comment (imjasonh) [inline cue.yaml:31]:
```suggestion
```

File: cue.yaml [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+    strip-prefix: v
+
+subpackages:
+  - name: cue-bash-completion
+    description: bash completion for cue
+    pipeline:
+      - runs: |
+          set -x
+          mkdir -p "${{targets.subpkgdir}}/usr/share/bash-completion"
+          mv "${{targets.destdir}}/usr/share/bash-completion/completions" \
```
Comment (imjasonh) [inline cue.yaml:47]: In addition to the suggested edit, these should run `./cue-cli completion foo > ${{targets.subpkgdir}}/usr/share/...`

File: Makefile (hunk 1 above)
Comment (developer-guy) [reply Makefile]: thanks, didn't know that, fixed.

**Review by imjasonh — DISMISSED** [review]

**Review by imjasonh — DISMISSED** [review]

File: cue.yaml [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+    strip-prefix: v
+
+subpackages:
+  - name: cue-bash-completion
+    description: bash completion for cue
+    pipeline:
+      - runs: |
+          set -x
+          mkdir -p "${{targets.subpkgdir}}/usr/share/bash-completion"
+          ${{targets.destdir}}/usr/bin/cue completion bash > ${{targets.destdir}}/usr/share/bash-completion/completions/cue
```
Comment (imjasonh) [inline cue.yaml:43]: This makes `[subpkgdir]/usr/share/...` then emits the output into `[destdir]/usr/share/...` -- should this be creating and writing to `subpkgdir`?

Can you test this change to ensure the apks that are produced include the expected output?

**Review by kaniini — DISMISSED** [review]

**Review by rawlingsj — APPROVED** [review]

**Pull request merged**